pkgs/applications/editors/lapce/default.nix

```diff
@@ -0,0 +1,69 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, cmake
+, pkg-config
+, python3Full
+, perl
+, freetype
+, fontconfig
+, libxkbcommon
+, xcbutil
+, libX11
+, libXcursor
+, libXrandr
+, libXi
+, vulkan-loader
+, autoPatchelfHook
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "lapce";
+  version = "0.0.10";
+
+  src = fetchFromGitHub {
+    owner = "lapce";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "tOVFm4DFQurFU4DtpPwxXQLbTGCZnrV1FfYKtvkRxRE=";
+  };
+
+  cargoPatches = [ ./fix-build.patch ];
+
+  cargoSha256 = "BwB3KgmI5XnZ5uHv6f+kGKBzpyxPWcoKvF7qw90eorI=";
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    python3Full
+    perl
+    autoPatchelfHook
```

Comment:
Why do we need this? Are we not building from source?

Reply:
We are, but there is a missing runtime dependency on `libvulkan.so.1`. We fix it by adding
`patchelf --add-needed  libvulkan.so.1 $out/bin/lapce` on `preFixup` and using `autoPatchelfHook` but maybe there is a better way.

Maybe making a wrapper that adds the proper path to `LD_LIBRARY_PATH` is more sensible?

Reply:
And，some dependencies(rust crates) need `perl` and `python3`. Like this:

https://github.com/alexcrichton/openssl-src-rs/blob/8a3f62066fb5d5243bd951faf4a6ad79501d03eb/src/lib.rs#L133

Reply:
> Maybe making a wrapper that adds the proper path to LD_LIBRARY_PATH is more sensible?

or only using patchelf. autoPatchelfHook is normally used with downloaded blobs only.

Reply:
```nix
  preFixup = ''
    patchelf --add-needed ${vulkan-loader}/lib/libvulkan.so.1 $out/bin/lapce
  '';
```

removed autoPatchelfHook